### ISSUE TITLE
Fix the commented test in newsletters branch

### ISSUE THREAD

**Issue by artpi** [Automattic/wp-calypso#70651]
In the CI, a test failed on php5.6 and php7.0.

I did not find a way to replicate that, so I commented out the test for the time being:

https://github.com/Automattic/jetpack/pull/26417/commits/fe7ceb6c2d844ae8f5dabcc7f6d42abfabaa471d

You can see the failed job here:
https://github.com/Automattic/jetpack/actions/runs/3591868662/jobs/6046915819

I don't know how to replicate PHP 5.6 nor 7.0 in docker. 

The scope of this task is to fix the test and underlying code.

**Comment by millerf** [comment]
Issue is:

  ```
PHP Strict Standards:  Static function Automattic\Jetpack\My_Jetpack\Product::get_name() should not be abstract in /tmp/wordpress-latest/src/wp-content/plugins/jetpack/jetpack_vendor/automattic/jetpack-my-jetpack/src/products/class-product.php on line 141
```

**Comment by millerf** [comment]
Issue seems to have been fixed on trunk. Rebasing fixed the issue.
Code is https://github.com/Automattic/jetpack/pull/27905

Slack discussion is p1670927103503079-slack-CDLH4C1UZ

**Comment by millerf** [comment]
Fixed p1670943048796509/1670927103.503079-slack-CDLH4C1UZ
